File: angry_professor/ap.js
const angryProfessor = (k, a) => {
    let onTime = 0;
    for (let i = 0; i < a.length; i++) {
        if (a[i] <= 0) {
            onTime++;
        }
    }
    return onTime >= k ? "NO" : "YES";
}


console.log(angryProfessor(3, [-1, -3, 4, 2]));
console.log(angryProfessor(2, [0, -1, 2, 1]));

File: angry_professor/ap.py
def angry_professor(k:int, a:list) -> str:
    """
    Given an array of integers representing the arrival times of each student,
    determine if the class is cancelled.
    """
    return "YES" if sum(map(lambda x: x<=0, a)) < k else "NO"


if __name__ == "__main__":
    print(angry_professor(3, [-1, -3, 4, 2]))
    print(angry_professor(2, [0, -1, 2, 1]))

